url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_AgeGenderBarChart.java

```diff
@@ -0,0 +1,479 @@
+        while (ite.hasNext()) {
+            age = ite.next();
+            pop = singleYearFemaleAgeInYearsPopulationCount_TreeMap.get(age);
+            if (age.intValue() > startAgeOfEndYearInterval) {
+                if (!reportedPenultimateGroup) {
+                    femaleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+                    maxPop = maxPop.max(popGroup);
+                    //ageGroup = ageGroup + ageInterval;
+                    popGroup = new BigDecimal(pop.toString());
+                    reportedPenultimateGroup = true;
+                }
+                ageGroup = startAgeOfEndYearInterval;
+                popGroup = popGroup.add(pop);
+                maxPop = maxPop.max(popGroup);
+                femaleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+            } else {
+                if (age > ageGroup + ageInterval) {
+                    femaleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+                    maxPop = maxPop.max(popGroup);
+                    ageGroup = ageGroup + ageInterval;
+                    popGroup = new BigDecimal(pop.toString());
+                } else {
+                    popGroup = popGroup.add(pop);
+                }
+            }
+        }
+        ageGroup = 0;
+        popGroup = BigDecimal.ZERO;
+        reportedPenultimateGroup = false;
+        ite = singleYearMaleAgeInYearsPopulationCount_TreeMap.keySet().iterator();
+        while (ite.hasNext()) {
+            age = ite.next();
+            pop = singleYearMaleAgeInYearsPopulationCount_TreeMap.get(age);
+            if (age.intValue() > startAgeOfEndYearInterval) {
+                if (!reportedPenultimateGroup) {
+                    maleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+                    maxPop = maxPop.max(popGroup);
+                    //ageGroup = ageGroup + ageInterval;
+                    popGroup = new BigDecimal(pop.toString());
+                    reportedPenultimateGroup = true;
+                }
+                ageGroup = startAgeOfEndYearInterval;
+                popGroup = popGroup.add(pop);
+                maxPop = maxPop.max(popGroup);
+                maleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+            } else {
+                if (age > ageGroup + ageInterval) {
+                    maleAgeInYearsPopulationCount_TreeMap.put(ageGroup, popGroup);
+                    maxPop = maxPop.max(popGroup);
+                    ageGroup = ageGroup + ageInterval;
+                    popGroup = new BigDecimal(pop.toString());
+                } else {
+                    popGroup = popGroup.add(pop);
+                }
+            }
+        }
+        result[0] = femaleAgeInYearsPopulationCount_TreeMap;
+        result[1] = maleAgeInYearsPopulationCount_TreeMap;
+        result[2] = maxPop;
+        return result;
+    }
+
+    private static Object[] getDefaultData(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 374-416, lines 417-459 into separate methods.